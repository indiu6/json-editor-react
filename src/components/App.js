import React, { useState, useEffect } from 'react';
import AppRouter from './Router';
import { authService } from '../myFirebase';

const App = () => {
  const [isInit, setIsInit] = useState(false);
  const [userObj, setUserObj] = useState(false);

  useEffect(() => {
    authService.onAuthStateChanged((user) => {
      if (user) {
        setUserObj({
          displayName: user.displayName,
          uid: user.uid,
          updateProfile: (args) => user.updateProfile(args),
        });
      } else {
        setUserObj(null);
      }
      setIsInit(true);
    });
  }, []);

  const refreshUser = () => {
    const user = authService.currentUser;
    setUserObj({
      displayName: user.displayName,
      uid: user.uid,
      updateProfile: (args) => user.updateProfile(args),
    });
  };

  return (
    <>
      {isInit ? (
        <AppRouter
          refreshUser={refreshUser}
          isLoggedIn={Boolean(userObj)}
          userObj={userObj}
        />
      ) : (
        'Initializing...'
      )}
    </>
  );
};

export default App;
