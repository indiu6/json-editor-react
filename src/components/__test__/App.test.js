import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import App from '../App';

describe('Test', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
