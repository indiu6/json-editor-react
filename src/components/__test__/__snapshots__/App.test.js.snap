// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should take a snapshot 1`] = `
<DocumentFragment>
  Initializing...
</DocumentFragment>
`;
